### PR TITLE
Fix crashed when compressing empty image data.

### LINKED ISSUE #70197: Executing `Image.compress_from_channels` crashes Godot
### Godot version

4.0.beta.custom_build. ae86d907e

### System information

Ubuntu 22.10 - AMD RX 470, Gnome shell 43 X11

### Issue description

When executing
```
extends Node
func _process(delta):
	var temp_variable14001 = Image.new()
	temp_variable14001.compress_from_channels(0, -48, 45.0935244560242)
```
Godot crashes:
```
thirdparty/etcpak/ProcessDxtc.cpp:928: runtime error: load of null pointer of type '__int128 unsigned'

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.beta.custom_build (ae86d907e75c3de9b039e7cc77e521bc9739e6dc)
Dumping the backtrace. Please include this when reporting the bug on: https://github.com/godotengine/godot/issues
[1] godot4s() [0x26fb17c] (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/crash_handler_linuxbsd.cpp:56)
[2] /lib/x86_64-linux-gnu/libc.so.6(+0x3bcf0) [0x7fcf22a3bcf0] (??:0)
[3] CompressDxt5(unsigned int const*, unsigned long*, unsigned int, unsigned long) (/home/runner/work/GodotBuilds/GodotBuilds/godot/thirdparty/etcpak/ProcessDxtc.cpp:928)
[4] _compress_etcpak(EtcpakType, Image*, float) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/etcpak/image_compress_etcpak.cpp:232)
[5] _compress_bc(Image*, float, Image::UsedChannels) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/etcpak/image_compress_etcpak.cpp:89)
[6] Image::compress_from_channels(Image::CompressMode, Image::UsedChannels, float) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/io/image.cpp:2601)
[7] void call_with_variant_args_ret_helper<__UnexistingClass, Error, Image::CompressMode, Image::UsedChannels, float, 0ul, 1ul, 2ul>(__UnexistingClass*, Error (__UnexistingClass::*)(Image::CompressMode, Image::UsedChannels, float), Variant const**, Variant&, Callable::CallError&, IndexSequence<0ul, 1ul, 2ul>) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/variant/binder_common.h:680 (discriminator 4))
[8] void call_with_variant_args_ret_dv<__UnexistingClass, Error, Image::CompressMode, Image::UsedChannels, float>(__UnexistingClass*, Error (__UnexistingClass::*)(Image::CompressMode, Image::UsedChannels, float), Variant const**, int, Variant&, Callable::CallError&, Vector<Variant> const&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/variant/binder_common.h:464)
[9] MethodBindTR<Error, Image::CompressMode, Image::UsedChannels, float>::call(Object*, Variant const**, int, Callable::CallError&) const (/home/runner/work/GodotBuilds/GodotBuilds/godot/./core/object/method_bind.h:465)
[10] Object::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object/object.cpp:733 (discriminator 1))
[11] Variant::callp(StringName const&, Variant const**, int, Variant&, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/variant/variant_call.cpp:1159)
[12] GDScriptFunction::call(GDScriptInstance*, Variant const**, int, Callable::CallError&, GDScriptFunction::CallState*) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript_vm.cpp:1556)
[13] GDScriptInstance::callp(StringName const&, Variant const**, int, Callable::CallError&) (/home/runner/work/GodotBuilds/GodotBuilds/godot/modules/gdscript/gdscript.cpp:1827)
[14] bool Node::_gdvirtual__process_call<false>(double) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/node.h:237 (discriminator 5))
[15] Node::_notification(int) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/node.cpp:57)
[16] Node::_notificationv(int, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/./scene/main/node.h:45 (discriminator 14))
[17] Object::notification(int, bool) (/home/runner/work/GodotBuilds/GodotBuilds/godot/core/object/object.cpp:792)
[18] SceneTree::_notify_group_pause(StringName const&, int) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:868)
[19] SceneTree::process(double) (/home/runner/work/GodotBuilds/GodotBuilds/godot/scene/main/scene_tree.cpp:466 (discriminator 4))
[20] Main::iteration() (/home/runner/work/GodotBuilds/GodotBuilds/godot/main/main.cpp:3192)
[21] OS_LinuxBSD::run() (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/os_linuxbsd.cpp:878)
[22] godot4s(main+0x58f) [0x26fab45] (/home/runner/work/GodotBuilds/GodotBuilds/godot/platform/linuxbsd/godot_linuxbsd.cpp:75)
[23] /lib/x86_64-linux-gnu/libc.so.6(+0x23510) [0x7fcf22a23510] (??:0)
[24] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x89) [0x7fcf22a235c9] (??:0)
[25] godot4s(_start+0x25) [0x26fa4f5] (??:?)
-- END OF BACKTRACE --
================================================================

```

This example was found by Godot fuzzer - [Qarminer](https://github.com/qarmin/Qarminer), so it is quite unlikelly that this code could be used in real project, but still this should be handled gracefully.

Memory leaks or asan backtraces are visibe when using Godot build with sanitizers support - https://github.com/qarmin/GodotBuilds/actions (linux -> linux-editor-sanitizers)

### Steps to reproduce

Above

### Minimal reproduction project

N/A

### PR DESCRIPTION
**Pull request by stmSi**
Fix: #70197

### CODE REVIEW

**Comment by fire** [comment]
Looks like a straightforward guard expression.

File: core/io/image.cpp [hunk 1]
```diff
@@ -2594,6 +2594,11 @@ Error Image::compress(CompressMode p_mode, CompressSource p_source, float p_loss
 }
 
 Error Image::compress_from_channels(CompressMode p_mode, UsedChannels p_channels, float p_lossy_quality) {
+	if (data.is_empty()) {
+		ERR_PRINT_ONCE_ED("Empty Image data.");
+		return ERR_INVALID_DATA;
+	}
```
Comment (kleonc) [inline core/io/image.cpp:2600]: Why print once? I'd go for:
```suggestion
	ERR_FAIL_COND_V(data.is_empty(), ERR_INVALID_DATA);
```
instead (not sure if a dedicated error message is needed, I think `data.is_empty()` condition is quite clear).

Comment (stmSi) [reply core/io/image.cpp:2600]: The error is spamming a lot inside `_process` (see original issue #70197) 
Although it doesn't make sense to compress image every frame.
image

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**